### PR TITLE
libsearpc: build python3 package

### PR DESCRIPTION
**Pull request by greizgh**
###### Motivation for this change

While attempting to package seafile, I was in need of the python3 version of libsearpc.
This change seems harmless since seafile-client use the native lib, not the python one.
The applet runs fine with this derivation.

I also took the opportunity to remove the patch and switch to `autoreconfHook` for building.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libsearpc/default.nix [hunk 1]
```diff
@@ -11,19 +12,15 @@ stdenv.mkDerivation rec {
     sha256 = "18i5zvrp6dv6vygxx5nc93mai2p2x786n5lnf5avrin6xiz2j6hd";
   };
 
-  patches = [ ./libsearpc.pc.patch ];
-
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ automake autoconf libtool python2Packages.python python2Packages.simplejson ];
+  buildInputs =
+    [ autoreconfHook python3Packages.python python3Packages.simplejson ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsearpc/default.nix:17]:
```suggestion
  buildInputs = with python3Packages; [ python simplejson ];
```

File: pkgs/development/libraries/libsearpc/default.nix [hunk 2]
```diff
@@ -11,19 +12,15 @@ stdenv.mkDerivation rec {
     sha256 = "18i5zvrp6dv6vygxx5nc93mai2p2x786n5lnf5avrin6xiz2j6hd";
   };
 
-  patches = [ ./libsearpc.pc.patch ];
-
   nativeBuildInputs = [ pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsearpc/default.nix:15]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/development/libraries/libsearpc/default.nix [hunk 3]
```diff
@@ -11,19 +12,15 @@ stdenv.mkDerivation rec {
...
 
-  postPatch = "patchShebangs autogen.sh";
-
-  preConfigure = "./autogen.sh";
-
   meta = with lib; {
     homepage = "https://github.com/haiwen/libsearpc";
-    description = "A simple and easy-to-use C language RPC framework (including both server side & client side) based on GObject System";
+    description =
+      "A simple and easy-to-use C language RPC framework (including both server side & client side) based on GObject System";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsearpc/default.nix:23]:
```suggestion
    description = "A simple and easy-to-use C language RPC framework based on GObject System";
```

File: pkgs/development/libraries/libsearpc/default.nix [hunk 4]
```diff
@@ -11,19 +12,15 @@ stdenv.mkDerivation rec {
...
     license = licenses.lgpl3;
     platforms = platforms.linux;
     maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsearpc/default.nix:26]: Maybe you want to add yourself as a maintainer?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by greizgh** [comment]
Super fast review @SuperSandro2000 ! Thanks for your feedback, I addressed your remarks.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119366` at 0adaf8e0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>libsearpc</li>
    <li>seafile-client</li>
    <li>seafile-shared</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl3` is a deprecated license, please check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/libsearpc/default.nix:22:5:
```
   |
22 |     license = licenses.lgpl3;
   |     ^
```
</li>
  </ul>
</details>

**Pull request merged**